src/Microsoft.Health.Fhir.Core/Features/Operations/DataConvert/ContainerRegistryTemplateProvider.cs

```diff
@@ -0,0 +1,105 @@
+﻿// -------------------------------------------------------------------------------------------------
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License (MIT). See LICENSE in the repo root for license information.
+// -------------------------------------------------------------------------------------------------
+
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Threading;
+using System.Threading.Tasks;
+using DotLiquid;
+using EnsureThat;
+using Microsoft.Extensions.Logging;
+using Microsoft.Health.Fhir.Core.Features.Operations.DataConvert.Models;
+using Microsoft.Health.Fhir.TemplateManagement;
+using Microsoft.Health.Fhir.TemplateManagement.Exceptions;
+using Microsoft.Health.Fhir.TemplateManagement.Models;
+
+namespace Microsoft.Health.Fhir.Core.Features.Operations.DataConvert
+{
+    public class ContainerRegistryTemplateProvider : IDataConvertTemplateProvider
+    {
+        private readonly IContainerRegistryTokenProvider _containerRegistryTokenProvider;
+        private readonly ITemplateSetProviderFactory _templateSetProviderFactory;
+        private readonly ILogger<ContainerRegistryTemplateProvider> _logger;
+
+        public ContainerRegistryTemplateProvider(
+            IContainerRegistryTokenProvider containerRegistryTokenProvider,
+            ITemplateSetProviderFactory templateSetProviderFactory,
+            ILogger<ContainerRegistryTemplateProvider> logger)
+        {
+            EnsureArg.IsNotNull(containerRegistryTokenProvider, nameof(containerRegistryTokenProvider));
+            EnsureArg.IsNotNull(templateSetProviderFactory, nameof(templateSetProviderFactory));
+            EnsureArg.IsNotNull(logger, nameof(logger));
+
+            _containerRegistryTokenProvider = containerRegistryTokenProvider;
+            _templateSetProviderFactory = templateSetProviderFactory;
+            _logger = logger;
+        }
+
+        /// <summary>
+        /// Fetch template collection from container registry or built-in archive
+        /// </summary>
+        /// <param name="templateCollectionReference">Should be in "<registryServer>/<imageName>:<imageTag>" or "<registryServer>/<imageName>@<imageDigest>"  format</param>
```

Comment:
>  format [](start = 156, length = 8)

super nit:
double space, and `.` in the end of sentence, pls

Reply:
Thanks. Fixed